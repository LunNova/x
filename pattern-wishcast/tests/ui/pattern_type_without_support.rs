// SPDX-FileCopyrightText: 2025 LunNova
//
// SPDX-License-Identifier: MIT

//! Test that creating pattern types for enums without pattern support produces helpful errors

#![feature(never_type)]

use pattern_wishcast::pattern_wishcast;

pattern_wishcast! {
	// This enum doesn't declare pattern support
	enum Value = {
		HostValue { value: String },
		TupleValue { elements: Vec<Self> },
	};

	// This should produce an error:
	type StrictValue = Value is HostValue(_);
}

fn main() {}
