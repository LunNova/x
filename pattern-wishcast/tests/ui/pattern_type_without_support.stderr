error: Cannot create pattern type for enum `"Value"`. You must declare the enum with pattern support: `enum "Value" is <P: PatternTrait> { ... }`
  --> tests/ui/pattern_type_without_support.rs:9:1
   |
9  | / pattern_wishcast! {
10 | |     // This enum doesn't declare pattern support
11 | |     enum Value = {
12 | |         HostValue { value: String },
...  |
17 | |     type StrictValue = Value is HostValue(_);
18 | | }
   | |_^
   |
   = note: this error originates in the macro `pattern_wishcast` (in Nightly builds, run with -Z macro-backtrace for more info)
